server/app/views/applicant/AddressCorrectionBlockView.java

```diff
@@ -221,4 +224,17 @@ private ButtonTag renderNextButton(Params params) {
         .withClasses(ApplicantStyles.BUTTON_BLOCK_NEXT)
         .withId("cf-block-submit");
   }
+
+  @Override
+  protected ATag renderPreviousButton(Params params) {
```

Comment:
How hard would it be to make the address correction stage be its own block, so that we don't need this override to begin with? It seems like a code smell to me to be working around this.

If that is 100% not possible for some reason, can we do this, to avoid the duplication introduced here?

```java
  @Override
  protected ATag renderPreviousButton(Params params) {
    // Set the block index to the next block, so that the renderPreviousButton
    // method will render the correct block.
    Params newParams =
        params.toBuilder().setBlockIndex(params.blockIndex()+1).build();
    return super.renderPreviousButton(newParams);
  }
```

(which would require adding a `toBuilder()` method ([directions](https://github.com/google/auto/blob/main/value/userguide/builders-howto.md#-initialize-a-builder-to-the-same-property-values-as-an-existing-value-instance)).

This is not tested by me, so double check my code please.

Reply:
Refactoring this to be a block is more of a lift than it's worth at this point I think (the original designs called for this to be a popup which I think is still the v2 goal), but I get your point about not wanting to repeat code so I'll try it out with calling `super`. Good suggestion, thank you!

Reply:
Gotcha, would you make an issue with the tech-debt tag to look into it in the future, and tag it in the comment?

Reply:
Will do. I'm guessing this will be part of a larger track of v2 work for address correction, so I'm waiting to hear back on that but I will update once I have that.

Reply:
Cool, no need to block this PR on that.

Reply:
It's still TBD what exactly v2 work will entail so I made a placeholder issue for this in the meantime: https://github.com/civiform/civiform/issues/4650